sdk/dotnet/Pulumi.Automation/PulumiFn.cs

```diff
@@ -14,6 +15,11 @@ namespace Pulumi.Automation
     /// </summary>
     public abstract class PulumiFn
     {
+        /// <summary>
+        /// Get or set the assembly that will be used for required plugin discovery.
+        /// </summary>
+        public Assembly? DiscoveryAssembly { get; set; }
```

Comment:
Thoughts on naming, or if this should even be exposed? I was thinking we should allow it to be overridden so the consumer can be explicit, which I think is especially important for the delegate-driven inline program.

Do we think that this should be an optional parameter on the `PulumiFn.Create(..)` factory methods, or is the fact that it can be overriden on this property sufficient?